### PR TITLE
Hotfix for weapons

### PR DESCRIPTION
**Pull request by Magnius001**
# Description

Small changes to weapons:
- Map should now spawn more weapons other than shotguns and knifes. 
- Revert back the starting weapons of default player from `shotgun` to `pistol`. 
- Buff `fnscar` and `pistol`. 
- Only range weapons will trigger update events to ammo display.

Fixes / Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Manual testing
- [x] Existing tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by manya-k — APPROVED** [review]
looks good!
works as expected 
no new code smell introduced

File: source/core/assets/configs/player.json [hunk 1]
```diff
@@ -9,5 +9,5 @@
   "textureFilename": "images/player/player.png",
   "textureAtlasFilename": "images/player/player.atlas",
   "melee": "melee:knife",
-  "ranged": "ranged:shotgun"
+  "ranged": "ranged:pistol"
```
Comment (vsinh14) [inline source/core/assets/configs/player.json:12]: leave shotgun for debugging I will be changing this with the balance patch later today

**Review by vsinh14 — COMMENTED** [review]
change back weapon then it can be accepted

**Comment by manya-k** [comment]
> change back weapon then it can be accepted

if the player starts off with shotgun, then why would they ever pick pistol or when will it ever be used

**Review by vsinh14 — APPROVED** [review]
Looks good

**Pull request merged**